Docs: Improve title and documentation for share shortened link in Explore (#36380)
grafana/grafana: docs/sources/explore/_index.md

> **Note:** Available in Grafana 7.3 and later versions.

> The Share shortened link capability allows you to create smaller and simpler URLs of the format /goto/:uid instead of using longer URLs with query parameters. To create a shortened link to the executed query, click the **Share** option in Explore toolbar. Any shortened links that are never used will be automatically deleted after 7 days.

Reviewer: ```suggestion
The Share shortened link capability allows you to create smaller and simpler URLs of the format /goto/:uid instead of using longer URLs with query parameters. To create a shortened link to the executed query, click the **Share** option in the Explore toolbar. A shortened link that is never used will automatically get deleted after seven (7) days.
```
